### ISSUE TITLE
Enable bracketed paste in terminals

### ISSUE THREAD

**Issue by agriffis** [syl20bnr/spacemacs#3064]
Bracketed paste allows terminals to behave like GUI with regard to preserving indentation:
* https://cirw.in/blog/bracketed-paste
* http://www.emacswiki.org/emacs/bracketed-paste.el

For example, here's some text:
```
[aron@jupiter ~]$ sudo lvs
  LV              VG            Attr       LSize  Pool             Origin           Data%  Meta%  Move Log Cpy%Sync Con
  root            jupiter       -wi-ao---- 30.00g
  swap            jupiter       -wi-ao----  7.62g
  lv_aron         vg_aron       Cwi-a-C--- 64.00g [lv_aron_cache]  [lv_aron_corig]  100.00 6.02            0.00
  lv_metis        vg_aron       Cwi-a-C--- 35.97g [lv_metis_cache] [lv_metis_corig] 100.00 3.05            0.15
  backup          vg_backup     -wi-a-----  3.64t
  jabberwock_root vg_jabberwock -wi-a----- 10.00g
  lv_europa       vg_jabberwock -wi-a----- 10.00g
  lv_themisto     vg_jabberwock -wi-a----- 10.00g
  oliva           vg_jabberwock -wi-a----- 14.94g
```

Note that all the lines except the first start with a couple spaces. When I paste this in spacemacs terminal, I get this:
```
[aron@jupiter ~]$ sudo lvs
  LV              VG            Attr       LSize  Pool             Origin           Data%  Meta%  Move Log Cpy%Sync Con
    root            jupiter       -wi-ao---- 30.00g
      swap            jupiter       -wi-ao----  7.62g
        lv_aron         vg_aron       Cwi-a-C--- 64.00g [lv_aron_cache]  [lv_aron_corig]  100.00 6.02            0.00
          lv_metis        vg_aron       Cwi-a-C--- 35.97g [lv_metis_cache] [lv_metis_corig] 100.00 3.05            0.15
            backup          vg_backup     -wi-a-----  3.64t
              jabberwock_root vg_jabberwock -wi-a----- 10.00g
                lv_europa       vg_jabberwock -wi-a----- 10.00g
                  lv_themisto     vg_jabberwock -wi-a----- 10.00g
                    oliva           vg_jabberwock -wi-a----- 14.94g
```

By "paste" I mean `C-S-v` which is gnome-terminal's paste, but I've also tried middle mouse button in xterm, and `:paste-buffer` in tmux (which also should work since tmux adds the bracketed paste sequences)

The following in `.spacemacs` fixes the problem for me:
```
(defun dotspacemacs/layers ()
   dotspacemacs-additional-packages
   '(
     bracketed-paste
     )
)

(defun dotspacemacs/config ()
  (require 'bracketed-paste)
  (bracketed-paste-enable)
)
```

It would be great to see this integrated directly into spacemacs!

**Comment by AlejandroCatalina** [comment]
I'm working on this one.

Does a new layer seem right to add bracketed-paste support? I don't find it a right layer, there is shell maybe, or +tools, but I think this package covers more user cases than just shell pasting.

Any thoughts on this?

I'll have something by tomorrow, I'd like to hear some suggestions.

**Comment by StreakyCobra** [comment]
It seems it only affects terminal instances (`add-hook 'tty-setup-hook …`), so I would add it to the `+distribution/spacemacs` layer (that's just my opinion). People can still exclude it if is creates problem.

**Comment by AlejandroCatalina** [comment]
Okay, that's a good approach. Thanks ;).

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
